ibis/expr/api.py

```diff
@@ -1869,6 +1866,31 @@ def _timestamp_time(arg):
 # ---------------------------------------------------------------------
 # Time API
 
+def between(arg, lower, upper, tz=None):
+    """
+    Check if the input expr falls between the lower/upper bounds
+    passed. Bounds are inclusive. All arguments must be comparable.
+
+    Parameters
+    ----------
+    lower : str, datetime.time
+    upper : str, datetime.time
+    tz : str, timezone, default None
+
+    Returns
+    -------
+    is_between : BooleanValue
+    """
+    lower = _ops.as_value_expr(lower)
+    upper = _ops.as_value_expr(upper)
```

Comment:
This shouldn't be necessary as the validation will call these where appropriate.